app/api/questions.js

```diff
@@ -0,0 +1,132 @@
+const router = require('express').Router({ mergeParams: true })
+const db = require('../models/index')
+const { Op, ValidationError } = require("sequelize");
+const questionService = require('../services/question_service')
+const { requireAuthentication } = require('../../lib/auth')
+const string_helpers = require('../../lib/string_helpers')
+
+// GET /courses/course_id/questions?search=""&page=""&perPage=""
+// gets all the questions for a given course
+router.get('/', requireAuthentication, async function (req, res, next) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+    // check to make sure the user is a teacher for the specified course
+    const enrollmentTeacher = await db.Enrollment.findOne({
+        where: { 
+            userId: user.id,
+            courseId: courseId,
+            role: 'teacher'
+        }
+    })
+
+    if (enrollmentTeacher) {
+        // we want search to have a value even if it is left out of the url
+        const search = (req.query.search != null) ? req.query.search : ""
+        // page and perPage are required query string parameters
+        const page = (req.query.page != null) ? parseInt(req.query.page) : null
+        const perPage = (req.query.perPage != null) ? parseInt(req.query.perPage) : null
+        if (page != null && perPage != null) {
+            try {
+                // get the questions needed
+                const questions = await db.Question.findAll({
```

Comment:
Does the query need to have an order specified so the offset works correctly?

Reply:
Yeah we need page and perPage so that the offset works correctly. Like you said though, we can add default values for that to handle the case there is no query string parameter for page/perPage. I can add that tn